core/src/main/java/edu/wpi/grip/core/Range.java

```diff
@@ -0,0 +1,94 @@
+package edu.wpi.grip.core;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Holds the lower and upper bounds of a range of numbers.
+ */
+public class Range {
+
+  private double min;
+  private double max;
+
+  /**
+   * Creates a new range with both bounds equal to zero.
+   */
+  public Range() {
+    this(0, 0);
+  }
+
+  /**
+   * Creates a new range with the given bounds.
+   *
+   * @param min the lower end of the range
+   * @param max the upper end of the range
+   *
+   * @throws IllegalArgumentException if min > max
+   */
+  public Range(double min, double max) {
+    checkArgument(min <= max, "Min must be <= max");
```

Comment:
Would probably be more usable if, instead of throwing checkArgument exceptions, this just did the intelligent thing and set min and max to the smallest and the largest, respectively.